### ISSUE TITLE
Performance problems in later pytorch versions

### ISSUE THREAD

**Issue by felmoreno1726** [CSAILVision/GazeCapture#34]
Hi, is there a known reason why the Pytorch version chosen is 0.4.1?

It seems that later versions of pytorch take ~20x longer in the computation of gradients (back-propagation). I wonder if this is a known issue and the main reason why this version of torch was chosen. I encountered this behavior because I need a later version of pytorch to get some extra features.

**Comment by erkil1452** [comment]
Hi,
0.4.1 was probably the most recent version when I originally wrote this. Since then people reported issues with newer pytorch so updated the code and requirements.txt to pytorch 1.1.0. I forgot to update the readme which I fixed now. Thanks. 

I am actually not aware of any slowdown in never versions. Do you have cuda and cudnn and all that working properly in the newer environment? In principle it could also be something with Variables. The old code still uses pytorch "Variables" which are now replaced by tensors with the tensor.require_grad = True property. You can try to rewrite that part (remove the Variables on L221-225 and just set the property on the tensor directly). But it seemed to work so who knows.

**Comment by felmoreno1726** [comment]
Thank you for your prompt response @erkil1452. I actually thought the same thing about the "Variables" and the requires_grad_ function. I did change them, but it didn't made any improvement in terms of performance. I was able to pin point that the performance decreases by some change introduced in torch version 1.3.0, but I haven't been able to determine what. Maybe I should take this to the pytorch community.

As for CUDA and CUDNN environment... I was able to reproduce the problem in two different machines, which have both different GPUs, cuda and cudnn versions.
